bitcoin/src/blockdata/witness.rs

```diff
@@ -54,17 +56,16 @@ impl Decodable for Witness {
         if witness_elements == 0 {
             Ok(Witness::default())
         } else {
-            let mut cursor = 0usize;
-            let mut last = 0usize;
-            let mut second_to_last = 0usize;
+            // Leave space at the head for element positions.
+            // We will rotate them to the end of the Vec later.
```

Comment:
I think we could avoid rotation if we used `VecDequeue`. The problem then though is indices will be added in reverse order. Rotating just them is still better than moving the entire vec. We could also store `VecDequeue` directly in `Witness` but we need the left part to be contiguous which I don't think `VecDequeue` can gurantee.